test/runtests.jl

```diff
@@ -275,6 +292,292 @@ end
         close(conn)
     end
 
+
+    @testset "Binary" begin
```

Comment:
How close is this test section to tests above? Could any of it be put in a loop over `(TEXT, BINARY)`? Especially around the data parsing, could it loop over the data and decide whether to mark it as broken/unimplemented depending on whether the type is part of an implemented list of types? This would help remove duplication of tests and avoid them getting out of sync.